src/proto/streams/prioritize.rs

```diff
@@ -528,6 +528,12 @@ impl Prioritize {
         tracing::trace!("poll_complete");
 
         loop {
+            // Create a new span at the start of the loop to avoid continuously
+            // adding events to the same span due to tracing in the loop, which
+            // would lead to an ongoing increase in memory usage.
+            let span = tracing::debug_span!("poll_complete_loop");
+            let _e = span.enter();
```

Comment:
nit, take it or leave it: this could be simplified to
```suggestion
            let _span = tracing::debug_span!("poll_complete_loop").entered();
```